lib/phoenix_ecto/sql/sandbox.ex

```diff
@@ -32,21 +48,101 @@ defmodule Phoenix.Ecto.SQL.Sandbox do
   """
 
   import Plug.Conn
+  alias Plug.Conn
+  alias Phoenix.Ecto.SQL.SandboxSupervisor
+
+  @doc """
+  Spawns a sandbox process to checkout a connection for a remote client.
+
+  ## Examples
+
+      iex> {:ok, _owner_pid, metdata} = checkout(MyApp.Repo)
+  """
+  def checkout(repo, opts \\ []) do
+    case Supervisor.start_child(SandboxSupervisor, [repo, self(), opts]) do
+      {:ok, owner} ->
+        metadata = metadata_for(repo, owner)
+        {:ok, owner, metadata}
+
+      {:error, reason} ->
+        {:error, reason}
+    end
+  end
+
+  @doc """
+  Checks in a sandbox owner process holding a connection for a remote client.
+
+  ## Examples
+
+      iex> {:ok, owner_pid, metdata} = checkout(MyApp.Repo)
+      iex> :ok = checkin(owner_pid)
+  """
+  def checkin(owner) when is_pid(owner) do
+    GenServer.call(owner, :checkin)
+  end
 
   def init(opts \\ []) do
-    Keyword.get(opts, :sandbox, Ecto.Adapters.SQL.Sandbox)
+    opts
+    |> Enum.into(%{
+      sandbox: Ecto.Adapters.SQL.Sandbox,
+      header: "user-agent",
+      checkout: nil,
+      checkin: nil
+    })
```

Comment:
I think this would be clearer if we built the data structure upfront instead of through multiple transformations:

```elixir
%{
  sandbox: Keyword.get(opts, :sandbox, Ecto.Adapters.SQL.Sandbox),
  header: Keyword.get(opts, :header, "user-agent"),
  checkin: get_path(opts[:checkin]),
  checkin: get_path(opts[:checkout]),
  repo: opts[:repo]
}
```